app/uk/gov/hmrc/nationaldutyrepaymentcenter/connectors/CreateCaseConnector.scala

```diff
@@ -48,11 +48,15 @@ class CreateCaseConnector @Inject()(
       http.POST[EISCreateCaseRequest, EISCreateCaseResponse](
         url,
         request,
-        eisApiHeaders(correlationId, config.eisEnvironment, config.eisAuthorizationToken)
+        EISApiHeaders(correlationId, config.eisEnvironment, config.eisAuthorizationToken) ++ MDTPTracingHeaders(hc)
       )(
         implicitly[Writes[EISCreateCaseRequest]],
         readFromJsonSuccessOrFailure,
-        hc.copy(authorization = None),
+        hc.copy(
+          authorization = None, // sent via EISApiHeaders
+          requestId = None, // sent via MDTPTracingHeaders (None here prevents sending duplicate header to internal stubs)
```

Comment:
Just an observation, but I'm more confused about this comment than anything else.  Should we be hacking the code to improve the experience when using our stubs?

Reply:
I was aiming for getting the behaviour (in terms of what headers are sent) on `QA -> EIS` and `local -> stubs` the same, to reduce having to think about differences in behaviour based on environment...

(...which is somewhat forced upon us by the underlying `HeaderCarrier` change whereby its _only_ used for "internal" requests...)

Do any of @SteveSugden , @lambertegbe1 , @njmoorewitchcat and @rorticus0502 have an opinion on which is preferred:
1. "expected" difference in what headers are sent when sending requests to stubs vs EIS
1. confusing comments/hacks in the code to enforce consistent behaviour when sending requests to stubs vs EIS

Reply:
I'd vote for 2.  But just have the comment `// sent via MDTPTracingHeaders` which is similar to the line above.
Would that reduce the chance of someone removing those two lines in future thinking "its only for stubs, lets get rid")?

Reply:
Updated this so that the `x-request-id` and `x-session-id` values in the header carrier are not manipulated - this is to avoid risk of regression in sending implicit auditing events